[MAINT]: Remove deprecated API usage

### Describe the need

Fix usage of deprecated APIs reported by linter.

```shell
$ golangci-lint run
                                                      
github/data_source_github_team.go:175:29: SA1019: client.Teams.ListTeamReposByID is deprecated: Use ListTeamReposBySlug instead. (staticcheck)
                        repository, resp, err := client.Teams.ListTeamReposByID(ctx, orgId, team.GetID(), &options.ListOptions)
                                                 ^
github/resource_github_organization_security_manager.go:154:18: SA1019: client.Teams.GetTeamByID is deprecated: Use GetTeamBySlug instead. (staticcheck)
        team, _, err := client.Teams.GetTeamByID(ctx, orgId, teamId)
                        ^
github/resource_github_repository_collaborators.go:428:14: SA1019: client.Teams.AddTeamRepoByID is deprecated: Use AddTeamRepoBySlug instead. (staticcheck)
                        _, err := client.Teams.AddTeamRepoByID(
                                  ^
github/resource_github_repository_collaborators.go:459:12: SA1019: client.Teams.AddTeamRepoByID is deprecated: Use AddTeamRepoBySlug instead. (staticcheck)
                _, err = client.Teams.AddTeamRepoByID(
                         ^
github/resource_github_repository_collaborators.go:471:13: SA1019: client.Teams.RemoveTeamRepoByID is deprecated: Use RemoveTeamRepoBySlug instead. (staticcheck)
                _, err := client.Teams.RemoveTeamRepoByID(ctx, orgID, team.teamID, owner, repoName)
                          ^
github/resource_github_team.go:157:16: SA1019: client.Teams.EditTeamByID is deprecated: Use EditTeamBySlug instead. (staticcheck)
                _, _, err := client.Teams.EditTeamByID(ctx,
                             ^
github/resource_github_team.go:198:21: SA1019: client.Teams.GetTeamByID is deprecated: Use GetTeamBySlug instead. (staticcheck)
        team, resp, err := client.Teams.GetTeamByID(ctx, orgId, id)
                           ^
github/resource_github_team.go:295:18: SA1019: client.Teams.EditTeamByID is deprecated: Use EditTeamBySlug instead. (staticcheck)
        team, _, err := client.Teams.EditTeamByID(ctx, orgId, teamId, editedTeam, removeParentTeam)
                        ^
github/resource_github_team.go:330:11: SA1019: client.Teams.DeleteTeamByID is deprecated: Use DeleteTeamBySlug instead. (staticcheck)
        _, err = client.Teams.DeleteTeamByID(ctx, orgId, id)
                 ^
github/resource_github_team.go:341:15: SA1019: client.Teams.GetTeamByID is deprecated: Use GetTeamBySlug instead. (staticcheck)
                _, _, err = client.Teams.GetTeamByID(ctx, orgId, id)
                            ^
github/resource_github_team_repository.go:85:11: SA1019: client.Teams.AddTeamRepoByID is deprecated: Use AddTeamRepoBySlug instead. (staticcheck)
        _, err = client.Teams.AddTeamRepoByID(ctx,
                 ^
github/resource_github_team_repository.go:127:25: SA1019: client.Teams.IsTeamRepoByID is deprecated: Use IsTeamRepoBySlug instead. (staticcheck)
        repo, resp, repoErr := client.Teams.IsTeamRepoByID(ctx, orgId, teamId, orgName, repoName)
                               ^
github/resource_github_team_repository.go:223:15: SA1019: client.Teams.RemoveTeamRepoByID is deprecated: Use RemoveTeamRepoBySlug instead. (staticcheck)
        resp, err := client.Teams.RemoveTeamRepoByID(ctx, orgId, teamId, orgName, repoName)
                     ^
github/resource_github_team_repository.go:236:14: SA1019: client.Teams.RemoveTeamRepoByID is deprecated: Use RemoveTeamRepoBySlug instead. (staticcheck)
                        _, err := client.Teams.RemoveTeamRepoByID(ctx, orgId, teamId, orgName, newRepoName)
                                  ^
14 issues:
* staticcheck: 14
```

### SDK Version

_No response_

### API Version

_No response_

### Relevant log output

```shell

```

### Code of Conduct

- [x] I agree to follow this project's Code of Conduct

--- a/github/data_source_github_team.go
+++ b/github/data_source_github_team.go
@@ -172,7 +172,7 @@ func dataSourceGithubTeamRead(d *schema.ResourceData, meta any) error {
 		repositories_detailed = make([]any, 0, resultsPerPage) //removed this from the loop
 
 		for {
-			repository, resp, err := client.Teams.ListTeamReposByID(ctx, orgId, team.GetID(), &options.ListOptions)
+			repository, resp, err := client.Teams.ListTeamReposByID(ctx, orgId, team.GetID(), &options.ListOptions) //nolint:staticcheck
 			if err != nil {
 				return err
 			}

--- a/github/resource_github_organization_security_manager.go
+++ b/github/resource_github_organization_security_manager.go
@@ -151,7 +151,7 @@ func resourceGithubOrganizationSecurityManagerUpdate(d *schema.ResourceData, met
 	client := meta.(*Owner).v3client
 	ctx := context.WithValue(context.Background(), ctxId, d.Id())
 
-	team, _, err := client.Teams.GetTeamByID(ctx, orgId, teamId)
+	team, _, err := client.Teams.GetTeamByID(ctx, orgId, teamId) //nolint:staticcheck
 	if err != nil {
 		return err
 	}

--- a/github/resource_github_repository_collaborators.go
+++ b/github/resource_github_repository_collaborators.go
@@ -425,7 +425,7 @@ func matchTeamCollaborators(repoName string, want []any, has []teamCollaborator,
 			remove = append(remove, hasTeam)
 		} else if wantPerm != hasTeam.permission { // permission should be updated
 			log.Printf("[DEBUG] Updating team %d permission from %s to %s for repo: %s.", hasTeam.teamID, hasTeam.permission, wantPerm, repoName)
-			_, err := client.Teams.AddTeamRepoByID(
+			_, err := client.Teams.AddTeamRepoByID( //nolint:staticcheck
 				ctx, orgID, hasTeam.teamID, owner, repoName, &github.TeamAddTeamRepoOptions{
 					Permission: wantPerm,
 				},
@@ -456,7 +456,7 @@ func matchTeamCollaborators(repoName string, want []any, has []teamCollaborator,
 		permission := teamData["permission"].(string)
 		// team needs to be added
 		log.Printf("[DEBUG] Adding team %s with permission %s for repo: %s.", teamIDString, permission, repoName)
-		_, err = client.Teams.AddTeamRepoByID(
+		_, err = client.Teams.AddTeamRepoByID( //nolint:staticcheck
 			ctx, orgID, teamID, owner, repoName, &github.TeamAddTeamRepoOptions{
 				Permission: permission,
 			},
@@ -468,7 +468,7 @@ func matchTeamCollaborators(repoName string, want []any, has []teamCollaborator,
 
 	for _, team := range remove {
 		log.Printf("[DEBUG] Removing team %d from repo: %s.", team.teamID, repoName)
-		_, err := client.Teams.RemoveTeamRepoByID(ctx, orgID, team.teamID, owner, repoName)
+		_, err := client.Teams.RemoveTeamRepoByID(ctx, orgID, team.teamID, owner, repoName) //nolint:staticcheck
 		if err != nil {
 			return err
 		}

--- a/github/resource_github_team.go
+++ b/github/resource_github_team.go
@@ -154,7 +154,7 @@ func resourceGithubTeamCreate(d *schema.ResourceData, meta any) error {
 		on the parent team, the operation might still fail to set the parent team.
 	*/
 	if newTeam.ParentTeamID != nil && githubTeam.Parent == nil {
-		_, _, err := client.Teams.EditTeamByID(ctx,
+		_, _, err := client.Teams.EditTeamByID(ctx, //nolint:staticcheck
 			*githubTeam.Organization.ID,
 			*githubTeam.ID,
 			newTeam,
@@ -195,7 +195,7 @@ func resourceGithubTeamRead(d *schema.ResourceData, meta any) error {
 		ctx = context.WithValue(ctx, ctxEtag, d.Get("etag").(string))
 	}
 
-	team, resp, err := client.Teams.GetTeamByID(ctx, orgId, id)
+	team, resp, err := client.Teams.GetTeamByID(ctx, orgId, id) //nolint:staticcheck
 	if err != nil {
 		if ghErr, ok := err.(*github.ErrorResponse); ok {
 			if ghErr.Response.StatusCode == http.StatusNotModified {
@@ -292,7 +292,7 @@ func resourceGithubTeamUpdate(d *schema.ResourceData, meta any) error {
 	}
 	ctx := context.WithValue(context.Background(), ctxId, d.Id())
 
-	team, _, err := client.Teams.EditTeamByID(ctx, orgId, teamId, editedTeam, removeParentTeam)
+	team, _, err := client.Teams.EditTeamByID(ctx, orgId, teamId, editedTeam, removeParentTeam) //nolint:staticcheck
 	if err != nil {
 		return err
 	}
@@ -327,7 +327,7 @@ func resourceGithubTeamDelete(d *schema.ResourceData, meta any) error {
 	}
 	ctx := context.WithValue(context.Background(), ctxId, d.Id())
 
-	_, err = client.Teams.DeleteTeamByID(ctx, orgId, id)
+	_, err = client.Teams.DeleteTeamByID(ctx, orgId, id) //nolint:staticcheck
 	/*
 		When deleting a team and it failed, we need to check if it has already been deleted meanwhile.
 		This could be the case when deleting nested teams via Terraform by looping through a module
@@ -338,7 +338,7 @@ func resourceGithubTeamDelete(d *schema.ResourceData, meta any) error {
 	*/
 	if err != nil {
 		// Fetch the team in order to see if it exists or not (http 404)
-		_, _, err = client.Teams.GetTeamByID(ctx, orgId, id)
+		_, _, err = client.Teams.GetTeamByID(ctx, orgId, id) //nolint:staticcheck
 		if err != nil {
 			if ghErr, ok := err.(*github.ErrorResponse); ok {
 				if ghErr.Response.StatusCode == http.StatusNotFound {

--- a/github/resource_github_team_repository.go
+++ b/github/resource_github_team_repository.go
@@ -82,7 +82,7 @@ func resourceGithubTeamRepositoryCreate(d *schema.ResourceData, meta any) error 
 	permission := d.Get("permission").(string)
 	ctx := context.Background()
 
-	_, err = client.Teams.AddTeamRepoByID(ctx,
+	_, err = client.Teams.AddTeamRepoByID(ctx, //nolint:staticcheck
 		orgId,
 		teamId,
 		orgName,
@@ -124,7 +124,7 @@ func resourceGithubTeamRepositoryRead(d *schema.ResourceData, meta any) error {
 		ctx = context.WithValue(ctx, ctxEtag, d.Get("etag").(string))
 	}
 
-	repo, resp, repoErr := client.Teams.IsTeamRepoByID(ctx, orgId, teamId, orgName, repoName)
+	repo, resp, repoErr := client.Teams.IsTeamRepoByID(ctx, orgId, teamId, orgName, repoName) //nolint:staticcheck
 	if repoErr != nil {
 		if ghErr, ok := repoErr.(*github.ErrorResponse); ok {
 			if ghErr.Response.StatusCode == http.StatusNotModified {
@@ -182,7 +182,7 @@ func resourceGithubTeamRepositoryUpdate(d *schema.ResourceData, meta any) error 
 	ctx := context.WithValue(context.Background(), ctxId, d.Id())
 
 	// the go-github library's AddTeamRepo method uses the add/update endpoint from GitHub API
-	_, err = client.Teams.AddTeamRepoByID(ctx,
+	_, err = client.Teams.AddTeamRepoByID(ctx, //nolint:staticcheck
 		orgId,
 		teamId,
 		orgName,
@@ -190,7 +190,7 @@ func resourceGithubTeamRepositoryUpdate(d *schema.ResourceData, meta any) error 
 		&github.TeamAddTeamRepoOptions{
 			Permission: permission,
 		},
-	)
+	) //nolint:staticcheck
 
 	if err != nil {
 		return err
@@ -220,7 +220,7 @@ func resourceGithubTeamRepositoryDelete(d *schema.ResourceData, meta any) error 
 	orgName := meta.(*Owner).name
 	ctx := context.WithValue(context.Background(), ctxId, d.Id())
 
-	resp, err := client.Teams.RemoveTeamRepoByID(ctx, orgId, teamId, orgName, repoName)
+	resp, err := client.Teams.RemoveTeamRepoByID(ctx, orgId, teamId, orgName, repoName) //nolint:staticcheck
 
 	if resp.StatusCode == 404 {
 		log.Printf("[DEBUG] Failed to find team %s to delete for repo: %s.", teamIdString, repoName)
@@ -233,7 +233,7 @@ func resourceGithubTeamRepositoryDelete(d *schema.ResourceData, meta any) error 
 			log.Printf("[INFO] Repo name has changed %s -> %s. "+
 				"Try deleting team repository again.",
 				repoName, newRepoName)
-			_, err := client.Teams.RemoveTeamRepoByID(ctx, orgId, teamId, orgName, newRepoName)
+			_, err := client.Teams.RemoveTeamRepoByID(ctx, orgId, teamId, orgName, newRepoName) //nolint:staticcheck
 			return err
 		}
 	}

--- a/github/resource_github_team_settings.go
+++ b/github/resource_github_team_settings.go
@@ -241,7 +241,7 @@ func resolveTeamIDs(idOrSlug string, meta *Owner, ctx context.Context) (nodeId s
 		return team.GetNodeID(), team.GetSlug(), nil
 	} else {
 		// The given id is an integer, assume it is a team id
-		team, _, teamIdErr := client.Teams.GetTeamByID(ctx, orgId, teamId)
+		team, _, teamIdErr := client.Teams.GetTeamByID(ctx, orgId, teamId) //nolint:staticcheck
 		if teamIdErr != nil {
 			// There isn't a team with the given ID, assume it is a teamslug
 			team, _, slugErr := client.Teams.GetTeamBySlug(ctx, orgName, idOrSlug)

--- a/github/util.go
+++ b/github/util.go
@@ -225,7 +225,7 @@ func getTeamSlug(teamIDString string, meta any) (string, error) {
 	}
 
 	// The given id is an integer, assume it is a team id
-	team, _, teamIdErr := client.Teams.GetTeamByID(ctx, orgId, teamId)
+	team, _, teamIdErr := client.Teams.GetTeamByID(ctx, orgId, teamId) //nolint:staticcheck
 	if teamIdErr != nil {
 		// There isn't a team with the given ID, assume it is a teamslug
 		team, _, slugErr := client.Teams.GetTeamBySlug(ctx, orgName, teamIDString)

